test/e2e/common/secrets.go

```diff
@@ -313,3 +317,80 @@ func doSecretE2EWithMapping(f *framework.Framework, mode *int32) {
 
 	f.TestContainerOutput("consume secrets", pod, 0, expectedOutput)
 }
+
+//Existence of a second secret with same name and different namespace shouldn't affect the mount
+func doSecretE2EForMountNameSpaceIssue(f *framework.Framework, defaultMode *int32) {
+	var (
+		name            = "secret-test-" + string(uuid.NewUUID())
+		volumeName      = "secret-volume"
+		volumeMountPath = "/etc/secret-volume"
+		namespace1      = f.Namespace
+		namespace2      *api.Namespace
+		err             error
+	)
+
+	if namespace2, err = f.CreateNamespace("secret-namespace", nil); err != nil {
+		framework.Failf("unable to create new namespace %s: %v", namespace2.Name, err)
+	}
+
+	secret1, secret2 := secretForTest(namespace1.Name, name), secretForTest(namespace2.Name, name)
```

Comment:
Contents of both secrets should be different so that if the wrong secret is mounted, it is obvious.

Also, verify second secret in a separate pod.